lib/pwa.js

```diff
@@ -336,8 +336,10 @@ exports._save = function(pwa) {
 
       // Validates the manifest or, reject and return errors in an array
       const res = libManifest.validateManifest(manifest.raw, pwa.manifestUrl, pwa.absoluteStartUrl);
-      pwa.manifest = res.length === 0 ? manifest : Promise.reject(res);
-
+      if (res.length !== 0) {
+        return Promise.reject(res);
```

Comment:
res contains the rejection information from the validator, I could add an additional message to it.